File: go/sdk/state/global_state.go
package state

import (
	"github.com/FatsharkStudiosAB/haja-workers/go/internal/basefunction"
	"github.com/FatsharkStudiosAB/haja-workers/go/internal/communication"
	"github.com/FatsharkStudiosAB/haja-workers/go/internal/dispatcher"
	"github.com/FatsharkStudiosAB/haja-workers/go/internal/grpccache"
	"github.com/FatsharkStudiosAB/haja-workers/go/internal/grpcstore"
	"github.com/FatsharkStudiosAB/haja-workers/go/internal/maps"
	"github.com/FatsharkStudiosAB/haja-workers/go/internal/rpc"
)

type GlobalState struct {
	GrpcCache        *grpccache.Client
	GrpcStore        *grpcstore.Client
	ServerName       string
	Functions        *maps.SafeFunctionMap[string, basefunction.FunctionInterface]
	ResponseHandlers *maps.SafeFunctionMap[string, chan *[]byte]
	RpcClient        *rpc.RpcClient
	ExecutionState   *maps.SafeFunctionMap[string, any]
	WorkflowComm     communication.WorkflowCommunicator
	Dispatcher       *dispatcher.Dispatcher
}

File: python/sdk/src/workers_core/state/global_state.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any, Dict, Optional

from workers_core.communication.grpc_communicator import GrpcCommunicator
from workers_core.dispatcher.dispatcher import Dispatcher
from workers_core.rpc.rpc import RpcClient  # will add later


@dataclass
class GlobalState:
    server_name: str
    workflow_comm: GrpcCommunicator
    dispatcher: Dispatcher
    rpc_client: Optional[RpcClient] = None
    grpc_cache: Any | None = None
    grpc_store: Any | None = None
    functions: Dict[str, Any] = field(default_factory=dict)
    response_handlers: Dict[str, Any] = field(default_factory=dict)
    execution_state: Dict[str, Any] = field(default_factory=dict)
